src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataDeltaLinkWrapper.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.OData;
+
+namespace Microsoft.AspNet.OData.Formatter.Deserialization
+{
+    /// <summary>
+    /// Encapsulates an <see cref="ODataResource"/> and the inner nested resource infos.
+    /// </summary>
+    public sealed class ODataDeltaLinkWrapper : ODataItemBase
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="ODataDeltaLinkWrapper"/>.
+        /// </summary>
+        /// <param name="item">The wrapped item.</param>
+        public ODataDeltaLinkWrapper(ODataDeltaLinkBase item)
+            : base(item)
+        {
+           
+        }
+
+   
+        /// <summary>
+        /// Gets the wrapped <see cref="ODataResource"/>.
+        /// </summary>
+        public ODataDeltaLinkBase DeltaLink
```

Comment:
>public ODataDeltaLinkBase DeltaLink [](start = 8, length = 35)

directly set as:

public ODataDeltaLinkBase DeltaLink {get;}

save the value in the ctor

Reply:
/// <summary>
        /// Initializes a new instance of <see cref="ODataDeltaLinkWrapper"/>.
        /// </summary>
        /// <param name="item">The wrapped item.</param>
        public ODataDeltaLinkWrapper(ODataDeltaLinkBase item)
            : base(item)
        {
            DeltaLink = item;
        }

        /// <summary>
        /// Gets the wrapped <see cref="ODataResource"/>.
        /// </summary>
        public ODataDeltaLinkBase DeltaLink { get; }

---
In reply to: [512300418](https://github.com/OData/WebApi/pull/2275#discussion_r512300418) [](ancestors = 512300418)

Reply:
:thumbsup:

---
In reply to: [512321098](https://github.com/OData/WebApi/pull/2275#discussion_r512321098) [](ancestors = 512321098,512300418)

Reply:
Storing it on the DeltaLinkWrapper would avoid the cast when accessing.  Probably not a big deal, but as long as it's immutable on the base class we should be able to cache it here.

---
In reply to: [518425182](https://github.com/OData/WebApi/pull/2275#discussion_r518425182) [](ancestors = 518425182,512321098,512300418)